contracts/PriceOracle/PriceOracle.sol

```diff
@@ -0,0 +1,214 @@
+    /// @param newPriceFeed The new price feed address the token config is being updated to
+    event PriceOracleAssetPriceFeedUpdated(address indexed cToken, address oldPriceFeed, address newPriceFeed);
+
+    /// @notice The event emitted when an asset is removed to the mapping
+    /// @param cToken cToken address that the config was removed for
+    /// @param baseUnit Number of smallest units of measurement in a single whole unit set in the removed config.
+    /// @param priceFeed Address price feed set in the removed config
+    event PriceOracleAssetRemoved(address indexed cToken, uint256 baseUnit, address priceFeed);
+
+    /// @notice The max decimals value allowed for price feed
+    uint8 internal constant MAX_DECIMALS = 72;
+
+    /// @notice The number of digits the price is scaled to before adjusted by the base units
+    uint8 internal constant PRICE_SCALE = 36;
+
+    /// @notice cToken address for config not provided
+    error MissingCTokenAddress();
+
+    /// @notice BaseUnit is missing or set to value 0
+    error InvalidBaseUnit();
+
+    /// @notice Price feed missing or duplicated
+    /// @param priceFeed Price feed address provided
+    error InvalidPriceFeed(address priceFeed);
+
+    /// @notice Config already exists
+    /// @param cToken cToken address provided
+    error DuplicateConfig(address cToken);
+
+    /// @notice Config does not exist in the mapping
+    /// @param cToken cToken address provided
+    error ConfigNotFound(address cToken);
+
+    /// @notice Decimals retrieved from the price feed is too large to use for formatting
+    /// @param decimals Decimals retrieve from the price feed
+    error DecimalsTooLarge(uint8 decimals);
+
+    /// @notice Price returned by the feed is negative
+    /// @param price Price returned from the price feed
+    error NegativePrice(int256 price);
+
+    /// @notice Same price feed as the existing one was provided when updating the price feed config
+    /// @param cToken cToken address that the price feed update is for
+    /// @param existingPriceFeed Price feed address set in the existing config
+    /// @param newPriceFeed Price feed address provided to update to
+    error UnchangedPriceFeed(address cToken, address existingPriceFeed, address newPriceFeed);
+
+    /**
+     * @notice Construct a Price Oracle contract for a set of token configurations
+     * @param configs The token configurations that define which price feed and base unit to use for each asset
+     */
+    constructor(TokenConfig[] memory configs) {
+        // Populate token config mapping 
+        for (uint i = 0; i < configs.length; i++) {
+            TokenConfig memory config = configs[i];
+            validateTokenConfig(config);
+            // Check if duplicate configs were submitted for the same cToken
+            if (tokenConfigs[config.cToken].cToken != address(0)) revert DuplicateConfig(config.cToken);
```

Comment:
Can this check be moved into `validateTokenConfig()`?

Reply:
Good catch! I've moved it in the latest commit